src/ray/object_manager/plasma/object_store.h

```diff
@@ -0,0 +1,74 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "ray/object_manager/plasma/common.h"
+#include "ray/object_manager/plasma/plasma.h"
+
+namespace plasma {
+
+// ObjectStore tracks object created over memory allocation.
+// It provides APIs to create/get/seal/delete objects.
+class ObjectStore {
+ public:
+  ObjectStore();
+
+  // Create a new object given allocation, object_info and source.
+  // Abort if such object with the same object_id already exists.
+  const LocalObject *CreateObject(Allocation allocation,
+                                  const ray::ObjectInfo &object_info,
+                                  plasma::flatbuf::ObjectSource source);
+
+  // Get a local object by object_id.
+  // Returns nullptr if such object doesn't exist.
+  const LocalObject *GetObject(const ObjectID &object_id) const;
```

Comment:
shared_ptr?

Reply:
yup I was planning to use [RAII](https://en.cppreference.com/w/cpp/language/raii) for LocalObject in follow up PRs, so we can use shared_ptr to replace our custom reference counting. 
One challenge is that the abort call breaks the contract, which could free the object even if we have multiple reference of the object.

Reply:
Hmm interesting, is it possible to remove abort and replace it with normal refcounting?

Reply:
yeah, it requires some change to store.cc so we can defer this to later PRs.  After discuss with @rkooo567, another challenge is that using shared_ptr might change how eviction policy works, specifically, the current way eviction policy get notified when ref-cout changes won't work with shared-ptr.